### ISSUE TITLE
Page titles are not read to screen readers

### ISSUE THREAD

**Issue by paul-geoghegan** [flutter/flutter#178024]
### Steps to reproduce

1. Add a title to a page
2. Run your flutter app,
3. Turn on a screen reader,
4. Hear that only the application name is spoken instead pf the page title.

### Expected results

The window should have the title read out instead of the application name like the <title> element from HTML.

### Actual results

Just the name of the app is read for example my_flutter_app

### Code sample

<details open><summary>Code sample</summary>

```dart
//I just used the namer app example here
    import 'package:english_words/english_words.dart';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => MyAppState(),
      child: MaterialApp(
        title: 'Namer App',
        theme: ThemeData(
          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepOrange),
        ),
        home: MyHomePage(),
      ),
    );
  }
}

class MyAppState extends ChangeNotifier {
  var current = WordPair.random();
  void getNext() {
    current = WordPair.random();
    notifyListeners();
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    var appState = context.watch<MyAppState>();
    var pair = appState.current;

    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            BigCard(pair: pair),
            SizedBox(height: 10),
            ElevatedButton(
              onPressed: () {
                appState.getNext();
              },
              child: Text('Next'),
            ),
          ],
        ),
      ),
    );
  }
}

class BigCard extends StatelessWidget {
  const BigCard({super.key, required this.pair});
  final WordPair pair;
  @override
  build(BuildContext context) {
    final theme = Theme.of(context);
    final style = theme.textTheme.displayMedium!.copyWith(
      color: theme.colorScheme.onPrimary,
    );

    return Card(
      color: theme.colorScheme.primary,
      child: Padding(
        padding: const EdgeInsets.all(20.0),
        child: Text(
          pair.asLowerCase,
          style: style,
          semanticsLabel: "${pair.first} ${pair.second}",
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.35.4, on Microsoft Windows [Version
    10.0.26220.6972], locale en-IE) [622ms]
    • Flutter version 3.35.4 on channel stable at
      C:\Users\Paul\Documents\Flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git      
    • Framework revision d693b4b9db (7 weeks ago), 2025-09-16 14:27:41
      +0000
    • Engine revision c298091351
    • Dart version 3.9.2
    • DevTools version 2.48.0
    • Feature flags: enable-web, enable-linux-desktop,
      enable-macos-desktop, enable-windows-desktop, enable-android,   
      enable-ios, cli-animations, enable-lldb-debugging
[√] Windows Version (Windows 11 or higher, 25H2, 2009) [1,398ms]
[√] Android toolchain - develop for Android devices (Android SDK
    version 35.0.0) [3.5s]
    • Android SDK at C:\Users\Paul\AppData\Local\Android\Sdk
    • Emulator version 35.2.10.0 (build_id 12414864) (CL:N/A)
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = C:\Users\Paul\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android
      Studio\jbr\bin\java
      This is the JDK bundled with the latest Android Studio
      installation on this machine.
      To manually set the JDK path, use: `flutter config
      --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build
      21.0.3+-12282718-b509.11)
    • All Android licenses accepted.
[√] Chrome - develop for the web [247ms]
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
[√] Visual Studio - develop Windows apps (Visual Studio Build Tools
    2022 17.12.3) [246ms]
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual
      Studio\2022\BuildTools
    • Visual Studio Build Tools 2022 version 17.12.35527.113
    • Windows 10 SDK version 10.0.22621.0
[√] Android Studio (version 2024.2) [21ms]
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      21.0.3+-12282718-b509.11)
[√] VS Code (version 1.105.1) [18ms]
    • VS Code at C:\Users\Paul\AppData\Local\Programs\Microsoft VS
      Code
    • Flutter extension version 3.120.0
[√] Connected device (3 available) [412ms]
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows
      [Version 10.0.26220.6972]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome    
      141.0.7390.123
    • Edge (web)        • edge    • web-javascript • Microsoft Edge   
      142.0.3595.53
[√] Network resources [432ms]
    • All expected network resources are available.
• No issues found!
```

</details>

**Comment by paul-geoghegan** [comment]
Sorry my mistake, this seems to be happening on desktop.

**Comment by chunhtai** [comment]
This title is used to named the web tab or OS app title, which not necessary visible in the app itself. In this case, announcing it may be make user more confused. I don't think this is a bug.

**Comment by chunhtai** [comment]
It seems like `Semantics(namesRoute: '')` is what you are looking for. This property is used when there a route is push or pop from the screen (for example dialog route sets this with on its title https://github.com/flutter/flutter/blob/1f69497f4fb6f140bd63f96689f5eea0a98acabb/packages/flutter/lib/src/material/dialog.dart#L852)

**Comment by chunhtai** [comment]
This looks like something that needs to be fixed in desktop embedding. The title widget sends SystemChrome.setApplicationSwitcherDescription which goes through flutter/platform channel.

**Comment by paul-geoghegan** [comment]
Just for your reference here are the corisponding guidelines: https://www.w3.org/WAI/WCAG22/Understanding/page-titled.html Although these are called web guidelines they do apply to desktop apps under the European Accessibility Act so this is actually a very important feature for any org wishing to be compliant which is a legal requirement.
